hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneFsDelete.java

```diff
@@ -139,10 +140,22 @@ protected void processPath(PathData item) throws IOException {
       if (moveToTrash(item) || !canBeSafelyDeleted(item)) {
         return;
       }
-      if (!item.fs.delete(path, deleteDirs)) {
-        throw new PathIOException(item.toString());
+      BasicRootedOzoneFileSystem.InnerDeleteResult innerDeleteResult = null;
+      if (item.fs.getUri().getScheme().equals(OZONE_OFS_URI_SCHEME)) {
+        BasicRootedOzoneFileSystem ofs = (BasicRootedOzoneFileSystem) item.fs;
+        innerDeleteResult = ofs.getDeleteResult(path, deleteDirs);
+      } else {
+        if (!item.fs.delete(path, deleteDirs)) {
+          throw new PathIOException(item.toString());
+        }
+      }
+      if (innerDeleteResult != null && innerDeleteResult.isPartiallyDeleted()) {
```

Comment:
Need to consider `innerDeleteResult.isSuccess()` as `false` also and throw exception to keep the behaviour same as before.